Training/Program.cs

```diff
@@ -0,0 +1,99 @@
+﻿// --------------------------------------------------------------------------------------------
+// Training ~ A training program for new joinees at Metamation, Batch - July 2023.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------------
+// A10.2: DoubleEndedQueue<T>
+// This is like a queue but we can add or remove i.e. enqueue or dequeue at both the ends at the head and the tail.
+// This should allow to add at the head or remove at the head and add to the tail or remove at the tail.
+// Implement it using an array of T.
+// --------------------------------------------------------------------------------------------
+
+namespace Training {
+   #region Program --------------------------------------------------------------------------------
+   /// <summary>Custom Double ended Queue<T> Program</summary>
+   internal class Program {
+      #region Methods -----------------------------------------------
+      /// <summary>Checks for the custom Double ended Queue<T></summary>
+      /// <param name="args">arguments</param>
+      static void Main (string[] args) {
+         DEndTQueue<int> Nos = new ();
+         Nos.RearEnqueue (1);
+         Nos.RearEnqueue (2);
+         Nos.RearEnqueue (5);
+         Nos.FrontEnqueue (3);
+         Nos.FrontEnqueue (4);
+         for (int i = 0; i < 5; i++) Console.WriteLine (Nos.FrontDequeue ());
+      }
+      #endregion
+   }
+   #endregion
+
+   #region Class Double ended TQueue --------------------------------------------------------------
+   /// <summary>Double Ended Queue<T></summary>
+   /// <typeparam name="T">Type of the Queue</typeparam>
+   public class DEndTQueue<T> {

```

Comment:
```suggestion
   public class TDQueue<T> {

```
Naming not good. We had TStack, TQueue etc so should have named like that.